lib/controllers/devices.dart

```diff
@@ -40,4 +40,35 @@ class Devices {
     });
     return device;
   }
+
+  /// Method to get deprovision a device from its device ID.
+  static Future<bool> deproviseDevice(
+      http.Client client, String authToken, String deviceId, int reason) async {
+    ApiCalls apiCall = ApiCalls(client);
+    bool action =
+        await apiCall.deprovision(authToken, deviceId, reason).catchError((e) {
+      throw e;
+    });
```

Comment:
Is it necessary to catch the error if we are going to throw it immediately after?

Reply:
+1

Reply:
Already removed it.

Reply:
Why are the methods in here static btw?  Looks like you are passing in the authToken and the http client to everything?  Why not just do this once and make an object you inject into your workflow?  You could use this setup:
https://pub.dev/packages/provider

That is a way of doing something akin to dependency injection.  Flutter only rebuilds from the current widget down on a state change, so if you inject the class into the ux heirachy it is not rebuilt unless there is a state change.  The state objects (in a statefulwidget) are not rebuilt at all.

Reply:
Done, thanks for the advice.